{{cookiecutter.github_repository}}/docs/backend/api_mixins.md

```diff
@@ -0,0 +1,23 @@
+# DRF helpers
+
+## Permissions Per Action Helper
+
+DRF allows us to set permissions on an entire viewset. 
+
+```
+class FooViewSet(viewsets.ModelViewSet):
+    permission_classes = [permissions.IsAuthenticated]
+```
+
+`PermissionPerAction` mixin available in the `base` module allows us to set permission classes for each of the actions (`list`, `retrieve`, `create`, `partial_update`, `update` or any other custom actions) on the viewset.
+
+```
+import project_name.base.api.mixins import PermissionPerAction
+
+class FooViewSet(PermissionPerAction, viewsets.ModelViewSet):
+    permissions_per_action = {
+        "create": [permissions.IsAuthenticated],
+        "list": [permissions.AllowAny]
+    }
+```
```

Comment:
If you are creating a doc instead of using docstring then you should include the doc for `MultiSerializerMixin` as well, or you can just have it in the docstring and refer a summary version here. We can also look for auto class doc plugins like sphinx but now sure if we have one.

Reply:
I was thinking of adding the docs for `MultipleSerializerMixin` after this gets merged to keep the PR atomic. I wasn't sure if everyone would agree to adding this mixin.

Reply:
haha. okay. I was also exploring [mkautodocs](https://github.com/tomchristie/mkautodoc) and it works great with only caveat is that it then requires you to the Mkdocs with PYTHON_PATH env set to current directory, which is understandable.

```
PYTHON_PATH=. mkdocs serve
```
or 
```
PYTHON_PATH=. mkdocs build
```